.ci/integTestGen/src/integTestGen.jl

```diff
@@ -343,7 +363,25 @@ if abspath(PROGRAM_FILE) == @__FILE__
 
     if !isempty(depending_pkg)
         for p in depending_pkg
-            generate_job_yaml!(p, target_branch, job_yaml, package_infos)
+            # Handles the case of merging in the main branch. If we want to merge in the main branch, 
+            # we do it because we want to publish the package. Therefore, we need to be sure that there 
+            # is an existing version of the dependent QED packages that works with the new version of 
+            # the package we want to release. The integration tests are tested against the development 
+            # branch and the release version.
+            #  - The dev branch version must pass, as this means that the latest version of the other 
+            #    QED packages is compatible with our release version.
+            #  - The release version can exist. If they exist, it means that we do not need to release
+            #    a new version of this package.
```

Comment:
```suggestion
            #  - The release version integration tests may or may not pass. 
            #    1. If all of these pass, we will not need to increase the minor version of this package. 
            #    2. If they do not all pass, the minor version must be increased and the failing packages 
            #    must also be released later with an updated compat entry.
            #    In either case the release can proceed, as the released packages will continue to work
            #    because of their current compat entries.
```